x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/XContentUtils.java

```diff
@@ -39,7 +39,7 @@ public static void moveToFirstToken(XContentParser parser) throws IOException {
     public static void positionParserAtTokenAfterField(XContentParser parser, String field, String errorMsgTemplate) throws IOException {
         XContentParser.Token token = parser.nextToken();
 
-        while (token != null && token != XContentParser.Token.END_OBJECT) {
+        while (token != null) {
```

Comment:
This was needed to correctly parse Azure OpenAI chat completions responses. I've added an explicit test, which would fail otherwise to `XContentUtilsTests`